src/nitypes/bintime/_timedelta.py

```diff
@@ -175,6 +183,16 @@ def _to_hightime_timedelta(self) -> ht.timedelta:
         ) >> _BITS_PER_SECOND
         return ht.timedelta(seconds=whole_seconds, yoctoseconds=yoctoseconds)
 
+    @staticmethod
+    def _check_int_range(value: int, min: int, max: int) -> None:
```

Comment:
I don't think we get any benefit out of making this a static method, as opposed to a free function.

https://github.com/ni/nitypes-python/blob/main/src/nitypes/_arguments.py and https://github.com/ni/nitypes-python/blob/main/src/nitypes/_exceptions.py are appropriate places for shared range-checking / error message code.

It would be reasonable to move the overflow error to _exceptions.py and keep the range checking here. You wouldn't really need a helper method to do the range check since it would be two lines:
```
if not (_UINT64_MIN <= fractional_seconds <= _UINT64_MAX):
    raise int_out_of_range(fractional_seconds, _UINT64_MIN, _UINT64_MAX)
```